media-service/src/main/java/org/arpha/service/helper/MediaServiceHelper.java

```diff
@@ -0,0 +1,56 @@
+package org.arpha.service.helper;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.sas.SasProtocol;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.tuple.Pair;
+import org.arpha.dto.media.enums.AccessType;
+import org.arpha.dto.media.request.FileUploadRequest;
+import org.arpha.exception.GenerateFileLinkException;
+import org.arpha.property.AzureStorageProperties;
+import org.arpha.repository.UserRepository;
+import org.arpha.utils.Boxed;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.time.OffsetDateTime;
+
+@Component
+@RequiredArgsConstructor
+public class MediaServiceHelper {
```

Comment:
Conisder splitting this class into 2 separate classes. As of now it kind of breaks the single responsibility, also Im not much of a fan of such "Helper" classes. For example, you can create **FileRequestValidator** class and put upload and generateLink method into some kind of **BlobService** class